### PR TITLE
Disposable email project.

### PR DESCRIPTION
**Pull request by pradzikowski**

### CODE REVIEW

**Comment by pradzikowski** [comment]
@adeptofvoltron i was end my WIP process. I will wait for your review

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+php:
+  - '7.1'
+  - nightly
+
+before_script:
+  - composer self-update
+  - composer install --prefer-dist --no-interaction
+
+script:
+  - phpunit
```
Comment (adeptofvoltron) [inline .travis.yml]: more likely to use `./vendor/bin/phpunit` than you will make sure the travis will run proper version of phpunit

File: src/Exception/DisposableEmailException.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PCF\DisposableEmail\Exception;
+
+class DisposableEmailException extends \Exception
+{}
```
Comment (adeptofvoltron) [inline src/Exception/DisposableEmailException.php]: Opening class brace must be on a line by itself
Closing class brace must be on a line by itself

Comment (adeptofvoltron) [reply src/Exception/DisposableEmailException.php]: similar on VerifyDomainException

File: src/Resource/BlockList.php [hunk 3]
```diff
@@ -0,0 +1,2187 @@
...
+zombie-hive.com
+zomg.info
+zp.ua
+zumpul.com
+zxcv.com
+zxcvbnm.com
+zzz.com
+LIST;
+
+}
```
Comment (adeptofvoltron) [inline src/Resource/BlockList.php]: line after closing brace

File: src/Resource/BlockList.php [hunk 4]
```diff
@@ -0,0 +1,2187 @@
...
+zoetropes.org
+zombie-hive.com
+zomg.info
+zp.ua
+zumpul.com
+zxcv.com
+zxcvbnm.com
+zzz.com
+LIST;
+
```
Comment (adeptofvoltron) [inline src/Resource/BlockList.php]: useless empty line

File: src/RegexVerifier.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    /**
+     * {@inheritdoc}
+     * This method will check whole lists
+     *
+     * @throws VerifyDomainException
+     */
+    public function verifyDomain(string $domain): int
+    {
+        if(empty($this->collection)) {
```
Comment (adeptofvoltron) [inline src/RegexVerifier.php]: space after if keyword

File: src/RegexVerifier.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            throw new VerifyDomainException('DomainCollection does not set');
+        }
+
+        if(empty($domain)) {
```
Comment (adeptofvoltron) [inline src/RegexVerifier.php]: space after if keyword

File: src/RegexVerifier.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+    /**
+     * @param array $list
+     * @param string $domain
+     *
+     * @return bool
+     */
+    private function pregList(array $list, string $domain): bool
+    {
+        if(empty($list)) {
```
Comment (adeptofvoltron) [inline src/RegexVerifier.php]: space after if keyword

File: src/RegexVerifier.php [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        $lists = [
+            self::DOMAIN_UNTRUSTED => $this->collection->getBlockedList(),
+            self::DOMAIN_TRUSTED   => $this->collection->getTrustedList(),
+            self::DOMAIN_KNOWN     => $this->collection->getKnownList(),
+        ];
+
+        foreach ($lists as $status => $list) {
+            if(true === $this->pregList($list, $domain)) {
```
Comment (adeptofvoltron) [inline src/RegexVerifier.php]: space after if keyword

File: src/RegexVerifier.php [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+     * @return bool
+     */
+    private function pregList(array $list, string $domain): bool
+    {
+        if(empty($list)) {
+            return false;
+        }
+
+        foreach ($list as $pattern) {
+            if(true == preg_match($pattern, $domain)) { //IHATETHISFUNCTIONBECAUSEITRETURNINTINTEADBOOLEAN
```
Comment (adeptofvoltron) [inline src/RegexVerifier.php]: space after if keyword

File: src/EmailVerifierInterface.php [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+declare(strict_types=1);
+
+namespace PCF\DisposableEmail;
+
+/**
+ * Interface EmailVerifierInterface
+ *
+ * @package PCF\DisposableEmail\Verifier
+ */
+Interface EmailVerifierInterface
```
Comment (adeptofvoltron) [inline src/EmailVerifierInterface.php]: `I` => `i`

File: src/ListedVerifier.php [hunk 11]
```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PCF\DisposableEmail;
```
Comment (adeptofvoltron) [inline src/ListedVerifier.php]: line after namespace

File: src/ListedVerifier.php [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+     * {@inheritdoc}
+     * This method will check whole lists
+     *
+     * @param string $domain
+     *
+     * @return int
+     */
+    public function verifyDomain(string $domain): int
+    {
+        if(empty($this->collection)) {
```
Comment (adeptofvoltron) [inline src/ListedVerifier.php]: space after if keyword

File: src/ListedVerifier.php [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+            throw new VerifyDomainException('DomainCollection does not set');
+        }
+
+        if(empty($domain)) {
```
Comment (adeptofvoltron) [inline src/ListedVerifier.php]: space after if keyword

File: src/ListedVerifier.php [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        $lists = [
+            self::DOMAIN_UNTRUSTED => $this->collection->getBlockedList(),
+            self::DOMAIN_TRUSTED   => $this->collection->getTrustedList(),
+            self::DOMAIN_KNOWN     => $this->collection->getKnownList(),
+        ];
+
+        foreach ($lists as $status => $list) {
+            if(empty($list)) {
```
Comment (adeptofvoltron) [inline src/ListedVerifier.php]: space after if keyword

File: src/ListedVerifier.php [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+                continue;
+            }
+
+            if(in_array($domain, $list)){
```
Comment (adeptofvoltron) [inline src/ListedVerifier.php]: space after if keyword and one before brace

File: test/RegexVerifierTest.php [hunk 16]
```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PCF\DisposableEmailTest;
+
```
Comment (adeptofvoltron) [inline test/RegexVerifierTest.php]: expected one blank line...not two

File: test/RegexVerifierTest.php [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+            [ $collection, 'domain1.test', RegexVerifier::DOMAIN_UNTRUSTED ],
+            [ $collection, 'domain2.test', RegexVerifier::DOMAIN_UNTRUSTED ],
+            [ $collection, 'domain3.test', RegexVerifier::DOMAIN_UNTRUSTED ],
+            [ $collection, 'domain4.test', RegexVerifier::DOMAIN_TRUSTED   ],
+            [ $collection, 'domain5.test', RegexVerifier::DOMAIN_TRUSTED   ],
+            [ $collection, 'domain6.test', RegexVerifier::DOMAIN_KNOWN     ],
+            [ $collection, 'domain7.test', RegexVerifier::DOMAIN_UNKNOWN   ],
+        ];
+    }
+}
```
Comment (adeptofvoltron) [inline test/RegexVerifierTest.php]: expected one blank line at the end of file

File: src/RegexVerifier.php (hunk 9 above)
Comment (adeptofvoltron) [inline src/RegexVerifier.php]: `HP Warning:  preg_match(): Delimiter must not be alphanumeric or backslash in /home/praca/Pulpit/DisposableEmail/src/RegexVerifier.php on line 83`

Comment (adeptofvoltron) [inline src/RegexVerifier.php]: useless comment

**Review by adeptofvoltron — CHANGES_REQUESTED** [review]
till `verifyDomain` is not static method, it would be nice(but not must have) to create Factry for Listed/RegexVerifier.

other hand you can create some static methods inside both methods...that allow you to verifyDomain. All dependiences are inside package.

File: src/Resource/AbstractResourceList.php [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+     * @var string
+     */
+    protected static $list = '';
+
+    /**
+     * @return array
+     */
+    public static function getList(): array
+    {
+        return explode(PHP_EOL, trim(static::$list));
```
Comment (adeptofvoltron) [inline src/Resource/AbstractResourceList.php]: topic to discuss: You can copy list from [source](https://github.com/martenson/disposable-email-domains) as a text. But in other hands, thi lib will be more efficient if we would allready keep list as array

File: src/RegexVerifier.php (hunk 9 above)
Comment (pradzikowski) [reply src/RegexVerifier.php]: stfu :D

File: .travis.yml (hunk 1 above)
Comment (pradzikowski) [reply .travis.yml]: Sure, I will fix this.

File: src/Exception/DisposableEmailException.php (hunk 2 above)
Comment (pradzikowski) [reply src/Exception/DisposableEmailException.php]: 👌

File: src/RegexVerifier.php (hunk 9 above)
Comment (pradzikowski) [reply src/RegexVerifier.php]: Could you show me you runtime code ? what pattern you use ?

File: src/Resource/AbstractResourceList.php (hunk 18 above)
Comment (pradzikowski) [reply src/Resource/AbstractResourceList.php]: I think about it and yes i will clone source files as original. Imo this is not reasonable to be afraid about perfomance and efficient in this case. Users and frameworks can cache this data in faster storages than static txt. 

Current frameworks have fine mechanism for caching static resources (ex. routing)for example [syfony](http://symfony.com/doc/current/components/dependency_injection/compilation.html)

**Comment by pradzikowski** [comment]
@adeptofvoltron Imo i dont need to create factories for these classes because their implementation deeply based on users code. Of course i can create factory only for our resources but i dont think we need this. Maybe i will create builder, but i dont promise.

**Comment by pradzikowski** [comment]
https://www.catgifpage.com/cute

@adeptofvoltron done :)

File: test/RegexVerifierTest.php (hunk 16 above)
Comment (pradzikowski) [reply test/RegexVerifierTest.php]: done

File: test/RegexVerifierTest.php (hunk 17 above)
Comment (pradzikowski) [reply test/RegexVerifierTest.php]: done

File: src/ListedVerifier.php (hunk 11 above)
Comment (pradzikowski) [reply src/ListedVerifier.php]: done

File: src/RegexVerifier.php (hunk 8 above)
Comment (pradzikowski) [reply src/RegexVerifier.php]: done

**Pull request closed without merging**